components/nuclio/invoker/component.yaml

```diff
@@ -0,0 +1,21 @@
+name: nuclio invoker
+description: invoke nuclio function.
+inputs:
+  - {name: Url, type: String, description: 'function URL/endpoint'}
+  - {name: Body, type: String, description: 'request body', default: ''}
+  - {name: Log level, type: String, description: 'log level', default: 'info'}
+  - {name: Method, type: String, description: 'HTTP method GET|POST|..', default: ''}
+outputs:
+  - {name: output, type: String, description: 'function output'}
+implementation:
+  container:
+    image: nuclio/invoker
+    command: [
+      invoke,
+      -a, {inputValue: Url},
+      -b, {inputValue: Body},
+      -m, {inputValue: Method},
+      -l, {inputValue: Log level},
+    ]
+    fileOutputs:
+      output: /tmp/output
```

Comment:
So we deploy a 'microservice' , and invoke it?

Reply:
@animeshsingh see above re `deploy`, `invoke` is calling the function (e.g. assume i have a function which does ETL or query data, i invoke it with the query and it process the data), the functions are auto-scaling (from zero or N) and will dynamically allocate resources. e.g. see [this function example](https://github.com/v3io/tutorials/blob/master/demos/stocks/read-stocks.ipynb) which reads stock trading data.

note that you dont have to use both, e.g. the pipeline may just deploy an inference function and the `invoke` will be done directly by an app client via HTTP or stream, e.g. see [this function notebook example](https://github.com/v3io/tutorials/blob/master/demos/image-classification/infer.ipynb) of TF serving, it can be deployed automatically after the model was generated and serve new requests, the Tag can be used for ingress policies (e.g. A/B testing)